### PR TITLE
feat(react): Update reset password react view to use gql

### PR DESCRIPTION
**Pull request by vbudhram**
## Because

- We need to add the functionality to initate password reset

## This pull request

- Adds the gql resolver to password reset react view

## Issue that this pull request solves

Closes: https://mozilla-hub.atlassian.net/browse/FXA-6123

## Checklist

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Other information (Optional)

This PR also removes the `canGoBack` logic from password reset. Using the browser back navigation does the same.

### CODE REVIEW

File: packages/fxa-settings/package.json
Comment (dschom) [inline packages/fxa-settings/package.json:46]: Maybe not relevant. I think this is what tripped up the build by the way... some sort of auto formatting appears to be takin place inside the docker container, and then we get a conflict here. I suspect husky is at fault. This was fixed recently, so a rebase on main should fix the issue.

File: packages/fxa-settings/src/pages/ResetPassword/ResetPassword/index.stories.tsx
Comment (LZoog) [inline packages/fxa-settings/src/pages/ResetPassword/ResetPassword/index.stories.tsx:20]: Did you mean to remove the `AppLayout` from the stories file?

File: packages/fxa-settings/src/pages/ResetPassword/ResetPassword/index.test.tsx
Comment (dschom) [inline packages/fxa-settings/src/pages/ResetPassword/ResetPassword/index.test.tsx:72]: This fails for me when I test. It appears that remember-pw-link is the DOM elements id. Maybe just using getByText is good enough?

File: packages/fxa-settings/src/pages/ResetPassword/ResetPassword/index.tsx
Comment (LZoog) [inline packages/fxa-settings/src/pages/ResetPassword/ResetPassword/index.tsx:188]: Do we no longer need the conditional checks?

File: packages/fxa-settings/src/pages/ResetPassword/ResetPassword/index.tsx
Comment (LZoog) [inline packages/fxa-settings/src/pages/ResetPassword/ResetPassword/index.tsx:48]: We're missing a few other events as well.

**Review by dschom — APPROVED** [review]
I did note one test failure, but the state of the react component is as expected. I think its just a matter of altering the call to screen. Once this is fixed, feel free to land. I'll go ahead and R+ so as not to block you.

File: packages/fxa-settings/src/pages/ResetPassword/ResetPassword/index.tsx
Comment (LZoog) [inline packages/fxa-settings/src/pages/ResetPassword/ResetPassword/index.tsx:5]: I thought in #14767 we moved all `pages/ResetPassword/ResetPassword` files into `pages/ResetPassword` but I think we missed these... if your PR diff remains small, you could maybe use `git mv` to help us out there?

Comment (vpomerleau) [reply packages/fxa-settings/src/pages/ResetPassword/ResetPassword/index.tsx:5]: They've been moved with a recent PR, looks like they're in the right spot on `main`

**Review by bcolsson — APPROVED** [review]

File: packages/fxa-settings/src/pages/ResetPassword/index.tsx
Comment (pdehaan) [inline packages/fxa-settings/src/pages/ResetPassword/index.tsx:29]: Q: should this be generic `string` or some enum like `MozServices`?

```sh
git grep -n "serviceName?" | cat
# packages/fxa-auth-server/test/local/senders/emails.ts:3008:function sesMessageTagsHeaderValue(templateName: string, serviceName?: any) {
# packages/fxa-settings/src/components/Ready/index.tsx:18:  serviceName?: MozServices;
# packages/fxa-settings/src/pages/InlineRecoverySetup/index.tsx:19:  serviceName?: MozServices;
# packages/fxa-settings/src/pages/InlineTotpSetup/index.tsx:21:  serviceName?: MozServices;
# packages/fxa-settings/src/pages/ResetPassword/AccountRecoveryConfirmKey/index.tsx:24:  serviceName?: MozServices;
# packages/fxa-settings/src/pages/ResetPassword/ResetPasswordConfirmed/index.tsx:12:  serviceName?: MozServices;
# packages/fxa-settings/src/pages/ResetPassword/ResetPasswordWithRecoveryKeyVerified/index.tsx:14:  serviceName?: MozServices;
# packages/fxa-settings/src/pages/ResetPassword/index.tsx:33:  serviceName?: MozServices;
# packages/fxa-settings/src/pages/Signin/SigninConfirmed/index.tsx:12:  serviceName?: MozServices;
# packages/fxa-settings/src/pages/Signin/SigninRecoveryCode/index.tsx:22:  serviceName?: MozServices;
# packages/fxa-settings/src/pages/Signin/SigninTotpCode/index.tsx:21:export type SigninTotpCodeProps = { email: string; serviceName?: MozServices };
# packages/fxa-settings/src/pages/Signin/index.tsx:20:  serviceName?: MozServices;
# packages/fxa-settings/src/pages/Signup/PrimaryEmailVerified/index.tsx:11:  serviceName?: MozServices;
# packages/fxa-settings/src/pages/Signup/SignupConfirmed/index.tsx:12:  serviceName?: MozServices;
# packages/fxa-settings/src/pages/Signup/index.tsx:28:  serviceName?: MozServices;
```

File: packages/fxa-settings/src/pages/ResetPassword/index.test.tsx
Comment (vpomerleau) [inline packages/fxa-settings/src/pages/ResetPassword/index.test.tsx:33]: What would you think about using `MOCK_ACCOUNT.primaryEmail.email` for testing?

File: packages/fxa-settings/src/pages/ResetPassword/index.test.tsx
Comment (vpomerleau) [inline packages/fxa-settings/src/pages/ResetPassword/index.test.tsx:67]: We have been preferring accessible tests wherever possible - this could be `screen.getByRole('link',{name:'Remember your password? Sign in'})` to also check that this uses the right element.

File: packages/fxa-settings/src/pages/ResetPassword/index.tsx
Comment (vpomerleau) [reply packages/fxa-settings/src/pages/ResetPassword/index.tsx:29]: Thanks @pdehaan, I think you're right that this should use use the `MozServices` enum, unless it will be a problem once we pull the service name from the relier model?

File: packages/fxa-settings/src/pages/ResetPassword/index.tsx
Comment (vpomerleau) [inline packages/fxa-settings/src/pages/ResetPassword/index.tsx:36]: There is a `viewName` declared on line 24 - should probably just keep one of the two?

File: packages/fxa-settings/src/pages/ResetPassword/index.tsx
Comment (vpomerleau) [inline packages/fxa-settings/src/pages/ResetPassword/index.tsx:89]: This error message should probably appear in a `Banner` component vs Input tooltip.

image

The tooltip is used if the email field is empty on submission:

image

File: packages/fxa-settings/src/pages/ResetPassword/index.tsx
Comment (vpomerleau) [inline packages/fxa-settings/src/pages/ResetPassword/index.tsx:46]: Just a note here that  @LZoog  and I spotted that `usePageViewEvent` is causing some problems as `usePageViewEvent` calls `useUserPreferences` which calls `useAccount`, and results in `TypeError: this.apolloClient.cache.readQuery(...) is null` when the page is loaded on localhost.

**Review by vpomerleau — COMMENTED** [review]
Thanks @vbudhram for working on this first major route setup! 
I've tested to see if the page loads as expected on localhost with the feature flag turned on. Initially had some issues because of `usePageViewEvent` that I worked around by replacing with the non-hook version `logPageViewEvent`. With that done, the page loaded as expected and redirected as expected on submission 🚀

File: packages/fxa-settings/src/pages/ResetPassword/en.ftl
Comment (vpomerleau) [inline packages/fxa-settings/src/pages/ResetPassword/en.ftl:16]: This string will need a new FTL id to trigger a localization update. Also, perhaps using `<link>` as an element name would be clearer for localizers? 

FWIW, the name used here can be anything, as long as it matches the name given in the `elems` attribute

File: packages/fxa-settings/src/pages/ResetPassword/index.tsx
Comment (vpomerleau) [inline packages/fxa-settings/src/pages/ResetPassword/index.tsx:118]: Should this be react-router's `<Link>` element? It's what we've been preferring, unless the link is external - then we use the `<LinkExternal>` component.

File: packages/fxa-settings/src/pages/ResetPassword/index.tsx
Comment (vpomerleau) [inline packages/fxa-settings/src/pages/ResetPassword/index.tsx:128]: The fallback string will need to include the full link element to render properly. One pattern that we have been using is creating the element in a `const`, then using it in both the `elems` and the fallback string.

File: packages/fxa-settings/src/pages/ResetPassword/index.tsx
Comment (vpomerleau) [inline packages/fxa-settings/src/pages/ResetPassword/index.tsx:96]: Should this type of error also go in the banner? The `errorText` passed to the input will show up in a tooltip. From what I see in the current content-server version, I think the tooltip is only used if the email input was left blank?

**Review by vpomerleau — COMMENTED** [review]
Thanks for checking on my comments! I have a few more comments about the Banner (thanks for adding it in!) - the localization there has been tricky and we've been learning from our experience with recent tickets.

Also - I saw that some `yarn.lock` updates are included here. Is that intentional? I've had some issues with those also popping up in a few of my other PRs yesterday.

File: packages/fxa-settings/src/models/Account.ts
Comment (LZoog) [inline packages/fxa-settings/src/models/Account.ts:1]: Small nit, guess we could add a license here since we've touched L1 🤷‍♀️

File: packages/fxa-settings/src/pages/ResetPassword/index.test.tsx
Comment (LZoog) [inline packages/fxa-settings/src/pages/ResetPassword/index.test.tsx:133]: I _think_ in your previous test we need `await act(async () => {` since you're checking function calls next but when you're checking for `getBy` stuff you can use `await findBy` instead:

```
fireEvent.click(screen.getByText('Begin Reset'));
await screen.findByText('Sign up');
```

File: packages/fxa-settings/src/pages/ResetPassword/index.tsx
Comment (LZoog) [inline packages/fxa-settings/src/pages/ResetPassword/index.tsx:42]: I'm guessing you intend to update this in other reset PW tickets assigned this sprint 👍

File: packages/fxa-settings/src/pages/ResetPassword/index.tsx
Comment (LZoog) [inline packages/fxa-settings/src/pages/ResetPassword/index.tsx:47]: I fix the issue with this [here](https://github.com/mozilla/fxa/pull/14844#discussion_r1098056157) but not sure it's the best long term solution. You could change this to `use` when you pull service name from the relier.

File: packages/fxa-settings/src/pages/ResetPassword/index.tsx
Comment (LZoog) [inline packages/fxa-settings/src/pages/ResetPassword/index.tsx:178]: Smol syntactic sugar, `{...{ errorText }}`

File: packages/fxa-settings/src/pages/ResetPassword/index.tsx
Comment (LZoog) [inline packages/fxa-settings/src/pages/ResetPassword/index.tsx:196]: If you click away from the input, there's overlap between the label and placeholder text. Not sure how we're handling this elsewhere without digging.

image

File: packages/fxa-settings/src/pages/ResetPassword/index.tsx
Comment (LZoog) [reply packages/fxa-settings/src/pages/ResetPassword/index.tsx:96]: Good call out, agreed this should render `<Banner>`.

**Review by LZoog — APPROVED** [review]
Have a couple call outs but going to r+ since bulk of work has been done. Thanks Vijay!

File: packages/fxa-settings/src/pages/ResetPassword/index.tsx
Comment (vpomerleau) [reply packages/fxa-settings/src/pages/ResetPassword/index.tsx:167]: I think this might be the only password input with a placeholder. Might be why it hasn't come up before. I'd be in favour of removing the placeholder for this input 🤭

**Comment by vbudhram** [comment]
Integration tests are failing from https://mozilla-hub.atlassian.net/browse/FXA-6780, merging this in to unblock remaining password reset flows.

**Pull request merged**
